lib/IRGen/IRGenModule.cpp

```diff
@@ -927,9 +929,9 @@ void IRGenModule::emitAutolinkInfo() {
                                        }),
                         AutolinkEntries.end());
 
-  if (TargetInfo.OutputObjectFormat == llvm::Triple::COFF ||
-      TargetInfo.OutputObjectFormat == llvm::Triple::MachO ||
-      Triple.isPS4()) {
+  if ((TargetInfo.OutputObjectFormat == llvm::Triple::COFF &&
+       !Triple.isOSCygMing()) ||
```

Comment:
MinGW/Cygwin's `ld` recognizes `.drectve ` section for export, but it is limited and is not same to MS  Linker.
It generated the warning "Warning: .drectve OOOO unrecognized" for "-lFoo" option in the section. We can see a part of ld source [here](https://fossies.org/linux/misc/binutils-2.26.1.tar.gz/binutils-2.26.1/ld/deffilep.y). (line 875~883, 934~950)